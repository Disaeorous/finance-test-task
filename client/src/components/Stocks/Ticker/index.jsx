export { Ticker } from './Ticker';
export { Headings } from './Headings';
