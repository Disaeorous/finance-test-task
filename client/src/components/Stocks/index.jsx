export { Ticker } from './Ticker';
export { Exchange } from './Exchange';
export { Stocks } from './Stocks';
