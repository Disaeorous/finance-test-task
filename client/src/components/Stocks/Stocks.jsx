import { Ticker } from './Ticker';
import { Exchange } from './Exchange';

export const Stocks = () => {
  return (
    <section className='stocks'>
      <Exchange>
        <Ticker />
      </Exchange>
    </section>
  );
};
