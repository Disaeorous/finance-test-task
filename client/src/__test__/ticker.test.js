import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { render } from '@testing-library/react';
import { Ticker } from '../components/Stocks';

describe('Ticker component', () => {
  it('Should render the Ticker component', () => {
    const props = {
      ticker: 'FFF',
      exchange: 2,
      price: 2,
      change: 2,
      change_percent: 2,
      dividend: 2,
      yield: 2,
      last_trade_time: 2,
    };

    const tickerComponent = shallow(<Ticker props={props} />);
		expect(tickerComponent).toMatchSnapshot();
  });
});
