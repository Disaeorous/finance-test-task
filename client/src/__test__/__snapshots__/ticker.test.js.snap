// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker component Should render the Ticker component 1`] = `ShallowWrapper {}`;
